spec/datadog/appsec/component_spec.rb

```diff
@@ -0,0 +1,49 @@
+# typed: ignore
+
+require 'datadog/appsec/spec_helper'
+require 'datadog/appsec/component'
+
+RSpec.describe Datadog::AppSec::Component do
+  describe '.build_appsec_component' do
+    context 'when appsec is enabled' do
+      it 'returns a Datadog::AppSec::Component instance' do
+        settings = double('AppSec::Settings', appsec: double('enabled', enabled: true))
+        component = described_class.build_appsec_component(settings)
+        expect(component).to be_a(described_class)
+      end
+    end
+
+    context 'when appsec is not enabled' do
+      it 'returns nil' do
+        settings = double('AppSec::Settings', appsec: double('enabled', enabled: false))
+        component = described_class.build_appsec_component(settings)
+        expect(component).to be_nil
+      end
+    end
+  end
+
+  describe '#shutdown!' do
+    context 'when processor is not nil' do
+      it 'finalize processor' do
+        settings = double('AppSec::Settings')
+        processor = double('AppSec::Processor')
+        component = described_class.new(settings)
+
+        expect(component).to receive(:processor).twice.and_return(processor)
```

Comment:
This seems to be a bit of a code smell of this class being a bit hard to test. Consider perhaps making the `Processor` a dependency that can be injected?

```ruby
def initialize(settings:, processor: Processor.new)
```

and then you can supply your own processor here. Furthermore, consider using a `instance_double` for the `Processor`, instead of just a naked `double`, since that means that means that RSpec will check that the class you mentioned actually exists and actually has the `finalize` method and that it doesn't receive arguments (aka that it exactly matches the double).